### ISSUE TITLE
Reset dialogue with keywords

### ISSUE THREAD

**Issue by basque21** [RasaHQ/rasa_core#740]
**Rasa Core version**: 0.9.7

**Python version**:  3.6

**Operating system** (windows, osx, ...): linux

**Issue**: Hi, I want to know if there is any possibility of reseting the dialogue if the user say -for example- "cancel". Say you are inside an action, the bot asks: "what type of cuisine" , but the user wants to exit and start again and say "cancel". How to check this and reset all slot values?

Thanks

**Content of domain file** (if used & relevant):
```yaml

```

**Comment by auzair92** [comment]
You could check the `rasa_core.events.AllSlotsReset` event. Documentation is available [here](https://core.rasa.com/api/events.html#rasa_core.events.AllSlotsReset).

Also, [this thread](https://github.com/RasaHQ/rasa_core/issues/370) might help, check it out too.

**Comment by basque21** [comment]
Thanks, I checked both links, but this is not what I'm wondering. I implemented the AllSlotsReset event when every conversation ends, but I want to exit from the conversation and end it with some keyword, before it is finished.

**Comment by auzair92** [comment]
So instead of implementing the `AllSlotsReset` event at the end of the conversation, you could implement it in the middle, where you want the conversation to stop. I guess you could implement a custom action (let's call it `ActionRestart`) for it and call that action when the intent `cancel` is recognized. Consider the following story flow.

```
## story_1
* greet
- utter_greet
> checkpoint_how_to_help
- utter_how_may_i_help

## story_2
- utter_what_cuisine
* cancel
- utter_okay_lets_restart
- ActionRestart
> checkpoint_how_to_help
```

You could train to recognize the `cancel` entity with training examples like 'cancel', 'let's restart', etc. ActionRestart will clear all for you and using [checkpoints](https://core.rasa.com/stories.html) you can start the flow of the conversation from any point that you'd like.

**Comment by basque21** [comment]
yes, I thought in this solution, but I don't want to create different story adding the possibility of cancelling in each step of the conversation, I want to recognize a special token and make the action "Restart" when the user says this special token...

**Comment by auzair92** [comment]
Then you could consider adding a stand-alone story like:
```
* cancel
- ActionRestart
```
Whenever the intent `cancel` is recognized, the flow shall be restarted in the `ActionRestart` method.
